admin/UcAdmin.php

```diff
@@ -125,37 +133,62 @@ public function add_uploadcare_js_to_admin($hook)
     public function uploadcare_handle()
     {
         $cdnUrl = $_POST['file_url'];
-        $id = $this->fileId($cdnUrl);
+        $uuid = $this->fileId($cdnUrl);
 
-        $file = $this->api->file()->fileInfo($id);
-        $attachment_id = $this->attach($file, null, $cdnUrl);
+        $file = $this->api->file()->fileInfo($uuid);
+        $modifiers = $_POST['uploadcare_url_modifiers'] ?? '';
+        if ($modifiers === 'null') {
+            $modifiers = '';
+        }
+
+        $attachment_id = $this->attach($file, $this->loadPostByUuid($uuid), [
+            'uploadcare_url_modifiers' => $modifiers,
+        ]);
 
         $result = [
             'attach_id' => $attachment_id,
             'fileUrl' => $cdnUrl,
             'isLocal' => false,
+            'uploadcare_url_modifiers' => $modifiers,
         ];
 
         echo \json_encode($result);
         \wp_die();
     }
 
+    public function loadPostByUuid(string $uuid): ?\WP_Post
+    {
+        global $wpdb;
+        $query = \sprintf('SELECT post_id FROM `%s` WHERE meta_value=\'%s\' AND meta_key=\'uploadcare_uuid\'', \sprintf('%spostmeta', $wpdb->prefix), $uuid);
```

Comment:
I'm not sure, but seems that query isn't being escaped. Perhaps `$uuid` value cannot be affected by user, but it's better to escape it IMHO.